### PR TITLE
Pre-release Revision

### PR DESCRIPTION
**Pull request by mecha**
See #2

### CODE REVIEW

File: src/AbstractIterator.php [hunk 1]
```diff
@@ -83,7 +83,7 @@ protected function _value()
      * Determines whether the current state of the iterator is valid.
      *
      * @since [*next-version*]
-     * @see Iterator::valid()
+     * @see   Iterator::valid()
```
Comment (XedinUnknown) [inline src/AbstractIterator.php:85]: There's no such code standard, AFAIK.

File: src/AbstractIterator.php [hunk 2]
```diff
@@ -130,5 +130,5 @@ protected function _valid()
      *
      * @return IterationInterface The iteration that represents the new state.
      */
-    abstract protected function _loop();
+    abstract protected function _nextIteration();
```
Comment (XedinUnknown) [inline src/AbstractIterator.php]: I was quite OK with the old method names, especially `_loop()`

File: src/AbstractIterator.php [hunk 3]
```diff
@@ -109,10 +109,8 @@ protected function _valid()
      * @since [*next-version*]
      *
      * @param IterationInterface|null $iteration The iteration to set.
-     *
-     * @throws InvalidArgumentException If not a valid iteration
      */
-    abstract protected function _setIteration(IterationInterface $iteration = null);
+    abstract protected function _setIteration($iteration);
```
Comment (XedinUnknown) [inline src/AbstractIterator.php]: 👍

File: src/AbstractRecursiveIterator.php [hunk 4]
```diff
@@ -38,8 +38,9 @@
      */
     protected function _construct()
     {
-        $this->_resetParents();
         parent::_construct();
+
+        $this->_resetParents();
```
Comment (XedinUnknown) [inline src/AbstractRecursiveIterator.php]: This is not a cleanup. You have switched the order of these operations. Why?

File: src/AbstractIterator.php (hunk 1 above)
Comment (mecha) [reply src/AbstractIterator.php:85]: You mean the indent? That's CS Fixer's doing.

File: src/AbstractRecursiveIterator.php (hunk 4 above)
Comment (mecha) [reply src/AbstractRecursiveIterator.php]: Isn't that how it should be? The parent's parameterless constructor should be called first when chaining, no?

Comment (XedinUnknown) [reply src/AbstractRecursiveIterator.php]: Not really. If there's a need for that, then yes. But I think that first, the class should do what it knows about, then let the ancestors do their thing. At least, that's how I do it everywhere.

File: src/AbstractIterator.php (hunk 1 above)
Comment (XedinUnknown) [reply src/AbstractIterator.php:85]: /shrug

File: src/AbstractRecursiveIterator.php (hunk 4 above)
Comment (mecha) [reply src/AbstractRecursiveIterator.php]: I see. Intuitively I'd think the opposite. Let ancestor do its thing, then the class gets its turn. I reverse them again?

Comment (XedinUnknown) [reply src/AbstractRecursiveIterator.php]: Reverse pls, so that at least it's consistent everywhere.

**Pull request merged**